### PR TITLE
DBZ-6899 Refactor Oracle streamings metrics

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-6899

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSourceMetrics.java [hunk 1]
```diff
@@ -0,0 +1,839 @@
...
+                ", abandonedTransactionIds=" + abandonedTransactionIds +
+                ", rolledBackTransactionIds=" + rolledBackTransactionIds +
+                "} " + super.toString();
+    }
+
+    /**
+     * Utility class for tracking histogram-based values for a duration-based metric.
+     */
+    @ThreadSafe
+    static class DurationHistogramMetric {
```
Comment (jpechane) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSourceMetrics.java:716]: There is `io.debezium.connector.spanner.metrics.latency` package. Could you please take a look if it makes sense to move those callses into the main repo and re-use them?

**Review by jpechane — APPROVED** [review]
@Naros Grate work thanks a lot. This LGTM for me. I ledt one comment related to statistics metrics calculation overlap with Spanner. Please take a look at it if it makes sense to work on re-use of the facilities provided. If not feel free to merge the PR yourself.

**Comment by Naros** [comment]
Hi @jpechane, I took a look at their implementation, and while there are some conceptual overlaps they also have this concept of `QuantileMeter` associated with each `Statistics` instance that tracks this "histogram" data, and each instance starts its own thread (about 11 in total) per Spanner connector.

Overall I think we can refactor this eventually, but I think that's likely better to coordinate with Nancy to understand the implications of doing that to make sure that both Oracle's needs and Google's are aligned when doing that refactor.  I have logged https://issues.redhat.com/browse/DBZ-6898.

**Comment by Naros** [comment]
Applied to main (=2.5).

**Pull request merged**